channeldb/channel.go

```diff
@@ -3121,6 +3121,28 @@ const (
 	Abandoned ClosureType = 5
 )
 
+// LocalForceCloseInfo encapsulates information that led to a channel being
+// force closed with the initiator being local. Currently local force closes
+// can be (1) purely user initiated, (2) automatic due to certain link errors,
+// or (3) automatic due to certain on-chain HTLC conditions. As such, only one
+// of the three fields in this struct will ever have meaningful information. But
+// this may be extended in the future. For example, we may decide to log HTLC
+// information that doesn't automatically result in a force close, but leaves
+// the channel in a state whereby the user has no option but to force close.
+type LocalForceCloseInfo struct {
```

Comment:
Could we use this in the `contractcourt` directly? I see that the HtlcActions isn't exactly the same type, but maybe we could convert? Otherwise we need encode/decode code in two places.
If we share this struct, it can have a public `Encode()` and `Decode()` method.